lib/Sema/ConstraintSystem.cpp

```diff
@@ -2934,8 +2976,16 @@ bool ConstraintSystem::diagnoseAmbiguityWithFixes(
     if (llvm::all_of(solutions, [&](const Solution &solution) {
           auto score = solution.getFixedScore();
           return score.Data[SK_Fix] == 0 && solution.Fixes.empty();
-        }))
+    })) {
+      SolutionDiff solutionDiff(solutions);
+
+      // We may have cases where we have ambiguity between two viable solutions
+      // on a generic type variable binding.
+      if (diagnoseConflictingGenericBinding(*this, solutionDiff, solutions))
```

Comment:
It would be great if we could unify this with `diagnoseConflictingGenericArguments` and call it before `diagnoseAmbiguityWithFixes` that way we can diagnose these conflicts regardless of fixes being present...

Reply:
Sure, I'll do in a moment :) 
Thanks @xedin

Reply:
One thing is that this is a bit more specific than `diagnoseConflictingGenericArguments`
because having no fixes and a generic conflict means we have more than one viable binding for this generic, whether `diagnoseConflictingGenericArguments` makes sense only when you do have `ArgumentMismatch` fixes. So, maybe make sense to keep this separated, WDYT?

Reply:
Let me know what you think, I can implement either way real quick :)

Reply:
Current logic simply limits this be used only when solutions have a certain fix but that's more of an artificial limitation than anything. It shouldn't matter whether there are any fixes or not if there is a conflict we should diagnose it.

Reply:
Great! I'll change it then :)

Reply:
Fixed :)